[build.webkit.org] Replace Trac with commitStore

JonWBedard opened the pull request:
#### f7c26dc807ef92a078756853af708f39a12c7f6a
<pre>
[build.webkit.org] Replace Trac with commitStore
<a href="https://rdar.apple.com/118421383">rdar://118421383</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=264842">https://bugs.webkit.org/show_bug.cgi?id=264842</a>

Reviewed by Alexey Proskuryakov.

Remove code using trac.webkit.org and replace it with a CommitStore based on commits.webkit.org
which allows callers to both query for and store commit objects themselves.

* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/BuildbotIteration.js:
(BuildbotIteration.prototype._deprecatedUpdateWithData): No need to propigate revision to Queue,
it already has the revision data.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/BuildbotQueue.js:
(BuildbotQueue.prototype.compareIterationsByRevisions): Use commitStore instead of trac object.
(BuildbotQueue.prototype.updateIterationPosition): Deleted.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/BuildbotQueueView.js:
(BuildbotQueueView.prototype._presentPopoverForRevisionRange): Use commitStore instead of trac object.
(BuildbotQueueView.prototype._revisionContentWithPopoverForIteration): Ditto.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/CommitStore.js: Added.
(CommitStore): Base class for all CommitSores.
(CommitStore.prototype.update): Update cached commit information for all branches.
(CommitStore.prototype.repr): Return the string representation of a commit.
(CommitStore.prototype.startPeriodicUpdates): Periodically run the &apos;update&apos; function.
(CommitStore.prototype.uuidFor): Convert commit object to UUID integer for sorting.
(CommitStore.prototype.addCommit): Add commit object to store.
(CommitStore.prototype.branchPosition): Return an integer indicating what position a commit is on
a branch. Note that for git repositories, this will be a position relative to the set of cached commits.
(CommitStore.prototype.nextRevision): Return the next-oldest commit for a commit ref.
(CommitStore.prototype.indexOfRef): Return local index of a specified commit.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Commits.js: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/CommitsWebKitOrg.js: Added.
(CommitsWebKitOrg.prototype.urlFor): Return commits.webkit.org URL for the specified ref.
(CommitsWebKitOrg.prototype.fetch): Start processing of fetch specified number commits from specified ref.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Initialization.js: Replace trac with commitStore.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Main.js: Replace trac with commitStore.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/QueueView.js:
(QueueView.prototype._appendPendingRevisionCount): Replace trac with commitStore, handle overflow from
for relativel branch positions.
(QueueView.prototype._popoverLinesForCommitRange): Consult commitStore to determine how to stringify a commit.
(QueueView.prototype._presentPopoverForPendingCommits): Ditto.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/ResultsDatabase.js:
(ResultsDatabase): Draft class to demonstrate how CommitStore protocol is extended.
(ResultsDatabase.prototype.urlFor): Return results database URL for the specified ref.
(ResultsDatabase.prototype.fetch): Start processing of fetch specified number commits from specified branch.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Trac.js: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/index.html: Remove Trac, include CommitStores.

Canonical link: <a href="https://commits.webkit.org/270793@main">https://commits.webkit.org/270793@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ec2db7ca783ac95ee438b2c50d05211334bfe32d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26436 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/5051 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/27696 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28602 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/24187 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/26782 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/6857 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/51/builds/2474 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/24165 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26696 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/6857 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/27696 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/3441 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/6857 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/27696 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29133 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/6857 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/27696 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/29754 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/3499 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/51/builds/2474 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/27648 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/26206 "Passed tests") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/4927 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/27696 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/3980 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [  ~~🛠 🧪 unsafe-merge~~](https://ews-build.webkit.org/#/builders/22/builds/3411 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/3821 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/BuildbotQueueView.js:
```diff
@@ -197,36 +197,38 @@ BuildbotQueueView.prototype = {
 
         if (!previousIteration)
             return content;
+        if (!commitStore)
```
Would we ever not have one?

JonWBedard replied:
On build.webkit.org we should always have one. But we don't yet have a `commitStore` for one of our buildbot instances, so it may be undefined/

JonWBedard committed:
[build.webkit.org] Replace Trac with commitStore
rdar://118421383
https://bugs.webkit.org/show_bug.cgi?id=264842

Reviewed by Alexey Proskuryakov.

Remove code using trac.webkit.org and replace it with a CommitStore based on commits.webkit.org
which allows callers to both query for and store commit objects themselves.

* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/BuildbotIteration.js:
(BuildbotIteration.prototype._deprecatedUpdateWithData): No need to propigate revision to Queue,
it already has the revision data.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/BuildbotQueue.js:
(BuildbotQueue.prototype.compareIterationsByRevisions): Use commitStore instead of trac object.
(BuildbotQueue.prototype.updateIterationPosition): Deleted.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/BuildbotQueueView.js:
(BuildbotQueueView.prototype._presentPopoverForRevisionRange): Use commitStore instead of trac object.
(BuildbotQueueView.prototype._revisionContentWithPopoverForIteration): Ditto.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/CommitStore.js: Added.
(CommitStore): Base class for all CommitSores.
(CommitStore.prototype.update): Update cached commit information for all branches.
(CommitStore.prototype.repr): Return the string representation of a commit.
(CommitStore.prototype.startPeriodicUpdates): Periodically run the 'update' function.
(CommitStore.prototype.uuidFor): Convert commit object to UUID integer for sorting.
(CommitStore.prototype.addCommit): Add commit object to store.
(CommitStore.prototype.branchPosition): Return an integer indicating what position a commit is on
a branch. Note that for git repositories, this will be a position relative to the set of cached commits.
(CommitStore.prototype.nextRevision): Return the next-oldest commit for a commit ref.
(CommitStore.prototype.indexOfRef): Return local index of a specified commit.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Commits.js: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/CommitsWebKitOrg.js: Added.
(CommitsWebKitOrg.prototype.urlFor): Return commits.webkit.org URL for the specified ref.
(CommitsWebKitOrg.prototype.fetch): Start processing of fetch specified number commits from specified ref.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Initialization.js: Replace trac with commitStore.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Main.js: Replace trac with commitStore.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/QueueView.js:
(QueueView.prototype._appendPendingRevisionCount): Replace trac with commitStore, handle overflow from
for relativel branch positions.
(QueueView.prototype._popoverLinesForCommitRange): Consult commitStore to determine how to stringify a commit.
(QueueView.prototype._presentPopoverForPendingCommits): Ditto.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/ResultsDatabase.js:
(ResultsDatabase): Draft class to demonstrate how CommitStore protocol is extended.
(ResultsDatabase.prototype.urlFor): Return results database URL for the specified ref.
(ResultsDatabase.prototype.fetch): Start processing of fetch specified number commits from specified branch.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Trac.js: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/index.html: Remove Trac, include CommitStores.

Canonical link: https://commits.webkit.org/270793@main
